werkzeug/http.py

```diff
@@ -212,7 +212,11 @@ def dump_options_header(header, options):
     segments = []
     if header is not None:
         segments.append(header)
-    for key, value in iteritems(options):
+    if isinstance(options, tuple):
+        iter_func = iter
+    else:
+        iter_func = iteritems
```

Comment:
~~~python
# in _compat
try:
    from collections.abc import Mapping
except ImporError:
    from collections import Mapping
~~~

~~~python
from werkzeug._compat import Mapping

iter_func = iteritems if isinstance(options, Mapping) else iter
~~~